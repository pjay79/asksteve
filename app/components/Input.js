import React from 'react';
import { TextInput, StyleSheet, Dimensions } from 'react-native';
import PropTypes from 'prop-types';

const Input = ({
  onChangeText, value, placeholder, secureTextEntry, style,
}) => (
  <TextInput
    placeholder={placeholder}
    returnKeyType="done"
    underlineColorAndroid="transparent"
    style={[styles.inputStyle, style]}
    onChangeText={onChangeText}
    value={value}
    secureTextEntry={secureTextEntry}
    autoCorrect={false}
    autoCapitalize="none"
    autoFocus
  />
);

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  inputStyle: {
    backgroundColor: 'white',
    paddingVertical: 15,
    paddingHorizontal: 10,
    width,
  },
});

Input.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  secureTextEntry: PropTypes.bool,
  style: PropTypes.shape(),
};

Input.defaultProps = {
  secureTextEntry: false,
  style: {},
};

export default Input;
