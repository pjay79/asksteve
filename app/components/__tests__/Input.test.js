import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';
import Input from '../Input';

describe('Input', () => {
  const mockFn = jest.fn();
  const mockProps = {
    onChangeText: mockFn,
    placeholder: 'Enter your username',
    value: 'Sally',
  };

  test('Renders correctly', () => {
    const tree = renderer.create(<Input {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Is working', () => {
    const { getByTestId } = render(<Input {...mockProps} />);
    const inputNode = getByTestId('input');
    expect(inputNode.value).toEqual('Sally');
  });
});
