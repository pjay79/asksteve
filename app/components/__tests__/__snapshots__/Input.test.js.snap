// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        autocapitalize="none"
        autocomplete="on"
        autocorrect="off"
        class="rn-MozAppearance-97bpaa rn-WebkitAppearance-n1uzsy rn-backgroundColor-14lw9ot rn-borderTopColor-nqhrom rn-borderRightColor-1878er4 rn-borderBottomColor-mbada rn-borderLeftColor-fpul1g rn-borderTopLeftRadius-ou6ah9 rn-borderTopRightRadius-t12b5v rn-borderBottomRightRadius-zmljjp rn-borderBottomLeftRadius-pm2fo rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-paddingTop-1qxgc49 rn-paddingRight-1qfoi16 rn-paddingBottom-19yat4t rn-paddingLeft-1hfyk0a rn-resize-1dz5y72 rn-width-16k0tzm rn-focusVisible-errtx7"
        data-focusable="true"
        dir="auto"
        placeholder="Enter your username"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </body>,
  "container": <div>
    <input
      autocapitalize="none"
      autocomplete="on"
      autocorrect="off"
      class="rn-MozAppearance-97bpaa rn-WebkitAppearance-n1uzsy rn-backgroundColor-14lw9ot rn-borderTopColor-nqhrom rn-borderRightColor-1878er4 rn-borderBottomColor-mbada rn-borderLeftColor-fpul1g rn-borderTopLeftRadius-ou6ah9 rn-borderTopRightRadius-t12b5v rn-borderBottomRightRadius-zmljjp rn-borderBottomLeftRadius-pm2fo rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-paddingTop-1qxgc49 rn-paddingRight-1qfoi16 rn-paddingBottom-19yat4t rn-paddingLeft-1hfyk0a rn-resize-1dz5y72 rn-width-16k0tzm rn-focusVisible-errtx7"
      data-focusable="true"
      dir="auto"
      placeholder="Enter your username"
      spellcheck="false"
      type="text"
      value=""
    />
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
